src/controls/TextInput.js

```diff
@@ -16,17 +18,75 @@ function TextInput(theme, skinName) {
     // show and load background image as skin (exploiting skin states)
     this.skinName = skinName || TextInput.SKIN_NAME;
     this._validStates = this._validStates || InputControl.stateNames;
+    this._type = TextInput.TEXT;
+    this.check = false;
 
     InputControl.call(this, theme);
 }
-
 TextInput.prototype = Object.create(InputControl.prototype);
 TextInput.prototype.constructor = TextInput;
+TextInput.uber = InputControl.prototype;
 module.exports = TextInput;
 
 // name of skin
 TextInput.SKIN_NAME = 'text_input';
 
+/**
+ * Text type: all input allowed
+ *
+ * @property TEXT
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.TEXT = 'text';
+
+/**
+ * Number type: only numbers are accepted
+ *
+ * @property NUMBER
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.NUMBER = 'number';
+
+/**
+ * names of possible types for a TextInput
+ *
+ * @property types
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.types = [
+  TextInput.TEXT, TextInput.NUMBER
+];
+
+/*
+ * set type attribute
+ */
+Object.defineProperty(TextInput.prototype, 'type', {
+    get: function () {
+        return this._type;
+    },
+    set: function (type) {
+        if (TextInput.types.indexOf(type) === -1) {
+            return;
+        }
+        this._type = type;
+        if (type === TextInput.NUMBER) {
+            this.createCheck();
+            if (this.text && !validators.isNumber(this.text)) {
+                this.text = '';
+            }
+            this.interceptors.push(validateTypeOfNumber);
```

Comment:
It seems that those for lines belong into createCheck as they create the check / validator / interceptor.

right now createCheck does nothing.